src/components/legend/draw.js

```diff
@@ -77,8 +77,39 @@ function drawOne(gd, opts) {
 
     var legendData;
     if(!inHover) {
-        if(!gd.calcdata) return;
-        legendData = fullLayout.showlegend && getLegendData(gd.calcdata, legendObj, fullLayout._legends.length > 1);
+        var calcdata = (gd.calcdata || []).slice();
+
+        var shapes = fullLayout.shapes;
+        for(var i = 0; i < shapes.length; i++) {
+            var shape = shapes[i];
+            if(!shape.showlegend) continue;
+
+            var shapeLegend = {
+                _fullInput: shape,
+                index: shape._index,
+                name: shape.name || shape.label.text || ('shape ' + shape._index),
+                legend: shape.legend,
+                legendgroup: shape.legendgroup,
+                legendgrouptitle: shape.legendgrouptitle,
+                legendrank: shape.legendrank,
+                legendwidth: shape.legendwidth,
+                showlegend: shape.showlegend,
+                visible: shape.visible,
+                opacity: shape.opacity,
+                mode: shape.type === 'line' ? 'lines' : 'markers',
+                line: shape.line,
+                marker: {
+                    line: shape.line,
+                    color: shape.fillcolor,
+                    symbol: shape.type === 'rect' ? 'square' : 'circle',
```

Comment:
Nice way to draw these, just pretending the shape is a scatter trace 🎉 

Path as circle is good if that's the only kind of shape you have, I just wonder if there's a reason to want to distinguish it from actual circle shapes? If so we could use maybe `'diamond'` for paths?

Reply:
Good call. Addressed in [d3004c1](https://github.com/plotly/plotly.js/pull/6653/commits/d3004c167895ca738dcb6f4df7ec0ef2a24500bb).

Reply:
`hexagon2` looks nice 🎉